numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -448,6 +450,42 @@ NPY_INPLACE @type@ npy_frexp@c@(@type@ x, int* exp)
 /**end repeat**/
 
 
+#ifdef powf
+#undef powf
+#endif
+#ifndef HAVE_POWF
+NPY_INPLACE npy_float npy_powf(npy_float x, npy_float y)
+{
+    return (npy_float) npy_pow((double)x, (double) y);
+}
+#endif
+
+#ifdef powl
+#undef powl
+#endif
+#ifndef HAVE_POWL
+#if defined(NPY_OS_DARWIN)
+NPY_INPLACE npy_longdouble npy_powl(npy_longdouble x, npy_longdouble y)
+{
+    if (x == 0 && y > 0) {
+        // Ensure that all the floating point exception flags are cleared.
+        // This shouldn't be necessary, but without it, unit tests on some
+        // of the Mac OSX CI platforms raise a divide-by-zero warning with
+        // a call such as `np.array([0, 1, 2], dtype=np.float128) ** 4`.
+        feclearexcept(FE_ALL_EXCEPT);
```

Comment:
Is this entirely without effect for future calcalations in C / Numpy?

Reply:
No, and that is a problem, because the internal ufunc code only checks the flags at the end of an inner loop.  For example, this should generate `RuntimeWarning: invalid value encountered in power`, but it does not:

```
In [24]: np.array([-1, 0], dtype=np.float128) ** 2.5
Out[24]: array([nan,  0.], dtype=float128)
```
With `float64`, the warning is generated:
```
In [25]: np.array([-1.0, 0.0]) ** 2.5
/Users/warren/mc37numpy/bin/ipython:1: RuntimeWarning: invalid value encountered in power
  #!/Users/warren/mc37numpy/bin/python
Out[25]: array([nan,  0.])
```
The call of `feclearexcept(FE_ALL_EXCEPT)` is a hack to get the tests on azure to pass.  The tests pass on my computer (Mac OSX 10.12.6) without that call, and I was surprised that they didn't pass on the azure Mac OSX test suite.  Somehow, even with the custom version of `npy_power`, calling `np.power` with `float128` data containing 0 and with positive exponents on the azure OSX platform results in a divide-by-zero warning.

P.S. I added "WIP" back to the PR title.

Reply:
The macos azure image seems to be [this one](https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/macos/macos-10.14-Readme.md) which is documented as using xcode 11.1 as default but also has other versions. Maybe the bug has been fixed in the version you are using?

Reply:
Thanks Matti.  My version is old: Xcode 9.2.  (I'm running OSX 10.12.6 on a 2013 Macbook Pro.  When I upgraded the OS earlier this year, I tried using a newer version of OSX but ran into problems.  10.12.6 has been working fine.)